dipy/io/tests/test_io_gradients.py

```diff
@@ -58,6 +59,14 @@ def test_read_bvals_bvecs():
     # You entered the bvecs on both sides:
     npt.assert_raises(IOError, read_bvals_bvecs, fbvecs, fbvecs)
 
+    # All possible delimiters should work
+    bv_file4 = tempfile.NamedTemporaryFile(mode='wt', delete=False)
```

Comment:
You can use `with` directive. it will help you to correctly close whatever the OS. To get an idea, look the test `test_fetch_data()` [here](https://github.com/nipy/dipy/blob/master/dipy/data/tests/test_fetcher.py#L91) and [here](https://github.com/nipy/dipy/blob/master/dipy/data/tests/test_fetcher.py#L66). I hope it will help you

Reply:
Can you provide an example on how to use `with` directive here? 
```Python
with tempfile.NamedTemporaryFile as temp:
    temp.write('4 44' 32')
    a, b = read_bvals_bvecs(temp.name, None)
```
I used the above but it always raises an error 😢